### ISSUE TITLE
Unit test build failure on AARCH64

### ISSUE THREAD

**Issue by elfringham** [tensorflow/tensorflow#55960]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

git HEAD

### Custom Code

No

### OS Platform and Distribution

CentOS 7

### Mobile device

n/a

### Python version

3.8.13

### Bazel version

5.1.1

### GCC/Compiler version

10.2.1

### CUDA/cuDNN version

n/a

### GPU model and memory

n/a

### Current Behaviour?

```shell
//tensorflow/compiler/mlir/xla/tests/hlo_to_lhlo_with_xla:hlo_text_to_lhlo_no_opt.hlotxt.test FAILED TO BUILD
```

### Standalone code to reproduce the issue

```shell
bazel test --test_timeout=300,500,-1,-1 --flaky_test_attempts=3 --test_output=all --cache_test_results=no --noremote_accept_cached --config=nonccl --copt="-mtune=generic" --copt="-march=armv8-a" --copt="-O3" --build_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --verbose_failures --build_tests_only -- //tensorflow/compiler/mlir/xla/tests/...
```

### Relevant log output

```shell
ERROR: /tmp/workspace/tensorflow-git/tensorflow/compiler/mlir/xla/BUILD:725:13: Linking tensorflow/compiler/mlir/xla/xla-opt-gpu failed: (Exit 1): gcc failed: error executing command 
  (cd /root/.cache/bazel/_bazel_root/db210f68f81d95ddcca9ae96b16ed72c/execroot/org_tensorflow && \
  exec env - \
    LD_LIBRARY_PATH=/opt/rh/devtoolset-10/root/usr/lib64:/opt/rh/devtoolset-10/root/usr/lib:/opt/rh/devtoolset-10/root/usr/lib64/dyninst:/opt/rh/devtoolset-10/root/usr/lib/dyninst:/usr/local/lib64 \
    PATH=/root/.cache/bazelisk/downloads/bazelbuild/bazel-5.1.1-linux-arm64/bin:/tmp/workspace/venv-cp38-cp38/bin:/opt/rh/devtoolset-10/root/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/tmp/workspace/venv-cp38-cp38/bin/python3 \
    PYTHON_LIB_PATH=/tmp/workspace/venv-cp38-cp38/lib/python3.8/site-packages \
    TF2_BEHAVIOR=1 \
  /opt/rh/devtoolset-10/root/usr/bin/gcc @bazel-out/aarch64-opt/bin/tensorflow/compiler/mlir/xla/xla-opt-gpu-2.params)
# Configuration: 92dba1cfc4f3c6543dbe3e81a2d486f27b48fb279788144993db9493fddedf49
# Execution platform: @local_execution_config_platform//:platform
/opt/rh/devtoolset-10/root/usr/bin/ld.gold: error: Stub is too far away, try a smaller value for '--stub-group-size'. The current value is 0x7ffbffc.
collect2: error: ld returned 1 exit status
INFO: Elapsed time: 291.246s, Critical Path: 156.86s
INFO: 916 processes: 917 local.
FAILED: Build did NOT complete successfully
```
</details>

**Comment by elfringham** [comment]
@cfRod @nSircombe

**Comment by elfringham** [comment]
Unfortunately simply reverting the above commit is not enough to fix the build as there is a second commit that also affects the same file.
https://github.com/tensorflow/tensorflow/commit/e90824d245eda658d3c42f72d7c72460a59b75a2

**Comment by joker-eph** [comment]
The error is one of binary getting to big to link in this code model. Any commit can trigger it really when we're close to the limit.

**Comment by elfringham** [comment]
Thanks @joker-eph . Any idea what should be done about it?

**Comment by elfringham** [comment]
This seems to have at least been alleviated by https://github.com/tensorflow/tensorflow/commit/cffb513f0f4401af29c4432f8206b58bcda24d20 as the test builds again now

**Comment by mohantym** [comment]
Hi @elfringham !
Could you confirm that the issue has been fixed in  2.10/2.11 version too. 
Thank you!
